src/main/kotlin/com/cultureamp/eventsourcing/RelationalDatabaseEventStore.kt

```diff
@@ -159,7 +159,12 @@ class RelationalDatabaseEventStore<M : EventMetadata> @PublishedApi internal con
             events
                 .select {
                     val eventTypeMatches = if (eventClasses.isNotEmpty()) {
-                        events.eventType.inList(eventClasses.map { eventTypeResolver.serialize(it.java) })
+                        val eventTypeDefinitions = eventClasses.map { eventTypeResolver.serialize(it.java) }
+                        val eventTypes = eventTypeDefinitions.eventTypes()
+                        val eventTypeClause = if (eventTypes.isNotEmpty()) events.eventType.inList(eventTypes) else Op.TRUE
```

Comment:
is the `else` possible? are you just being defensive here?

Reply:
It would be possible for someone to implement some type of shell EventTypeResolve that just returns empty everything because they don't wanna do any filtering